docs/api_reference/index.md

```diff
@@ -0,0 +1,32 @@
+
+The following is the API Reference for SmartX:
```

Comment:
Would it be a bit more discoverable if we have this structure in the navigation (`mkdocs.yml>nav`) instead?

Reply:
<img width="738" alt="image" src="https://github.com/splunk/addon-factory-smartx-ui-test-library/assets/11015487/a56d3fe7-d459-49d8-aad6-a667e5ab3380">

Reply:
I had mimicked the current doc of this framework, that it expands when clicked, but I think we don't have it for `.md` files. I have updated the nav to contain all.